### PR TITLE
Generic Benchmark API

### PR DESCRIPTION
**Pull request by kevinaer**
Generic benchmarking APIs for use with different frameworks.

Includes:
- BenchmarkResults class representing benchmark results, based on class in deepsparse
- Abstract BenchmarkRunner, which would be implemented using different SparseML frameworks
- Serialization via BenchmarkInfo
- Ability to run benchmarks via CLI
- Ability to run benchmarks via python API
- Ability to reload a BenchmarkInfo json and create a new run with the same conditions

### CODE REVIEW

File: src/sparseml/benchmark/info.py [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+        progress_steps = self.warmup_iterations + self.iterations
+        _LOGGER.debug("running {} items through model".format(progress_steps))
+        data_iter = (
+            enumerate(data)
+            if not show_progress
+            else auto.tqdm(enumerate(data), desc=desc, total=progress_steps)
+        )
+        for index, batch in data_iter:
+            if index < self.warmup_iterations:
+                continue
```
Comment (markurtz) [inline src/sparseml/benchmark/info.py:177]: we should still be running the batch for warmup here, correct?

**Review by markurtz — COMMENTED** [review]
I'm a bit confused on the separation of the pydantic schemas and the result objects. Is there any reason we can implement the result objects as pydantic models and override accordingly? Ideally would make the flow a lot easier so serialization is built into every return type we have (the original intention with everything being added in this new phase 2 step)

File: src/sparseml/benchmark/info.py (hunk 1 above)
Comment (kevinaer) [reply src/sparseml/benchmark/info.py:177]: Ah sorry was supposed to run self.run_batch then continue, will fix

**Review by markurtz — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]
GG!

**Pull request merged**